### PR TITLE
webUI: fix config route

### PR DESCRIPTION
**Pull request by andrevmatos**
With this PR, reverting back some changes made in 94a5c3cf, which avoided `--eth-rpc-address` being correctly passed to the Rest API, for providing `config.production.json` webUI config file with correct web3 address.
For that, changed `split_endpoint` to support getting `host` and `port` from URLs with protocol-prefixes (`http://`, `https://`) and path sufixes.
Last, testing for web3 endpoint on webUI, and falling back to `localhost:8545` in case of failure (e.g.: testnet).

### CODE REVIEW

File: raiden/api/rest.py [hunk 1]
```diff
@@ -185,6 +185,11 @@ def _serve_webui(self, file='index.html'):
             assert file
             web3 = self.flask_app.config.get('WEB3_ENDPOINT')
             if web3 and 'config.' in file and file.endswith('.json'):
+                if any(h in web3 for h in ('localhost', '127.0.0.1', '::1')) and\
```
Comment (hackaugusto) [inline raiden/api/rest.py]: please, don't use multiline if statements.

Comment (andrevmatos) [reply raiden/api/rest.py]: And what to do about to long line?

Comment (konradkonrad) [reply raiden/api/rest.py]: compile your conditional beforehand:
```python
my_condition = all(
  first_condition,
  second_condition,
)

if my_condition:
  ...
```

Comment (andrevmatos) [reply raiden/api/rest.py]: Ok.. thank you, @konradkonrad , will keep that in mind on future commits, but for now, managed to shrink this line.

File: raiden/utils/__init__.py [hunk 2]
```diff
@@ -62,8 +62,11 @@ def host_port_to_endpoint(host, port):
 
 
 def split_endpoint(endpoint):
-    host, port = endpoint.split(':')[:2]
-    port = int(port)
+    match = re.match(r'(?:[a-z0-9]*:?//)?([^:/]+)(?::(\d+))?', endpoint, re.I)
+    if not match:
+        raise ValueError('Invalid endpoint', endpoint)
+    host, port = match.groups()
+    port = port and int(port)
```
Comment (ulope) [inline raiden/utils/__init__.py]: This is a bit too non-obvious for my taste. 
```python
if port:
    port = int(port)
```
Would make the intent much clearer.

File: raiden/api/rest.py [hunk 3]
```diff
@@ -190,6 +190,11 @@ def _serve_webui(self, file='index.html'):
             assert file
             web3 = self.flask_app.config.get('WEB3_ENDPOINT')
             if web3 and 'config.' in file and file.endswith('.json'):
+                host = request.headers.get('Host')
+                if any(h in web3 for h in ('localhost', '127.0.0.1')) and host:
+                    _, _port = split_endpoint(web3)
+                    _host, _ = split_endpoint(host)
+                    web3 = "http://{}:{}".format(_host, _port)
```
Comment (ulope) [inline raiden/api/rest.py]: I'm not sure this will be useful in practice. We can't just assume that the eth node will be available (and accept connections) on port 8545 of whatever the `Host` header points to.

Comment (andrevmatos) [reply raiden/api/rest.py]: Yes, I agree it's not trustable, but it's the only information we have about the relation of the client and the server (in cases where client is not from localhost). It should cover one or other corner case, but the main part is the test-and-fallback on the web3 connection directly on webUI.

File: raiden/utils/__init__.py (hunk 2 above)
Comment (andrevmatos) [reply raiden/utils/__init__.py]: Done.

File: raiden/api/rest.py (hunk 3 above)
Comment (ulope) [reply raiden/api/rest.py]: Right, with the fallback it shouldn't hurt.
